SofaKernel/modules/SofaBaseMechanics/MechanicalObject.inl

```diff
@@ -2607,9 +2598,12 @@ void MechanicalObject<DataTypes>::buildIdentityBlocksInJacobian(const sofa::help
     cMatrix->endEdit();
 
 }
-#endif
-
 
+template <class DataTypes>
+void MechanicalObject<DataTypes>::renumberConstraintId(const sofa::helper::vector<unsigned>& /*renumbering*/)
+{
+    this->serr << "MechanicalObject<DataTypes>::renumberConstraintId not implemented in the MatrixDeriv constraint API" << this->sendl;
```

Comment:
No more serr...and more fundamentally... a pure virtual method is ...pure specifically to force the object inheriting from the parent to not forget to implement them. Providing an implementation like this one sounds like skipping over the safeguard that was put in place. Don't we have a better way to handle those cases ?

Reply:
If the override is mandatory, a pure virtual function is the way to go. This compiler-bypassing behavior should not be allowed.

If the override is not mandatory, then non-virtual + msg_warning("You didn't override") seems nice.

Reply:
I agree with your Guillaume :)